src/rdb.c

```diff
@@ -2271,7 +2271,7 @@ int rdbLoadRio(rio *rdb, int rdbflags, rdbSaveInfo *rsi) {
             sdsfree(key);
             decrRefCount(val);
         } else {
-            robj keyobj;
+            robj* keyobj = createRawStringObject(sdsdup(key), sdslen(key));
```

Comment:
since this event is rarely used, converting this object to a heap allocation is undesired.
instead we should force the module to use RM_CreateStringFromString.

BTW, FYI: #7528